src/main/webapp/app/exam/manage/exam-management.service.ts

```diff
@@ -61,7 +62,14 @@ export class ExamManagementService {
         const options = createRequestOption({ withStudents, withExerciseGroups });
         return this.http
             .get<Exam>(`${this.resourceUrl}/${courseId}/exams/${examId}`, { params: options, observe: 'response' })
-            .pipe(map((res: EntityResponseType) => ExamManagementService.convertDateFromServer(res)));
+            .pipe(map((res: EntityResponseType) => ExamManagementService.convertDateFromServer(res)))
+            .pipe(
+                tap((res: EntityResponseType) => {
+                    if (res.body) {
+                        this.accountService.setAccessRightsForCourse(res.body);
```

Comment:
Here you apply course rights to an exam. If this is intended, the method should be renamed to something like setAccessRightsForCourseOrExam